aten/src/ATen/quantized/Quantizer.cpp

```diff
@@ -171,30 +187,109 @@ T quantize_val(double scale, int64_t zero_point, float value) {
   int64_t qvalue;
   constexpr int64_t qmin = std::numeric_limits<typename T::underlying>::min();
   constexpr int64_t qmax = std::numeric_limits<typename T::underlying>::max();
-  checkZeroPoint<typename T::underlying>("quantize_val", zero_point);
-  qvalue = static_cast<int64_t>(std::nearbyint(value / scale + zero_point));
+  qvalue = static_cast<int64_t>(Round(value / scale + zero_point));
   qvalue = std::max<int64_t>(qvalue, qmin);
   qvalue = std::min<int64_t>(qvalue, qmax);
   return static_cast<T>(qvalue);
 }
 
 template <typename T, int precision>
 void quantize_vec(double scale, int64_t zero_point, const float *src, T *dst, size_t count) {
+  checkZeroPoint<typename T::underlying>("quantize_val", zero_point);
   for (int64_t i = 0; i < count; ++i) {
     dst[i] = quantize_val<T>(scale, zero_point, src[i]);
   }
 }
 
+#ifdef GEMMLOWP_NEON
+#define INT8_NEON_SIMD
+#endif
+
+#if defined(__SSE4_2__) && defined(__clang__)
+#include "NEON_2_SSE.h"
+#define INT8_NEON_SIMD
+#endif
+inline uint8_t Uint8Quantize(const float scale, const int32_t zero_point, const float value) {
+  const int32_t qmin = std::numeric_limits<uint8_t>::min();
+  const int32_t qmax = std::numeric_limits<uint8_t>::max();
+  auto r = zero_point + static_cast<int32_t>(Round(value / scale));
```

Comment:
Couple of questions:
1. Does this match what we do in FBGEMM? i.e adding zero-point after rounding:
2. We should add a test to ensure that Quantize and DeQuantize are identical across backends. I know they mismatched before, but lets change this to come up with a bit-exact match.

Reply:
No this is a difference right now. FBGEMM does `f = static_cast<int64_t>(Round(value / scale + zero_point));`
caffe2 Int8Quantize does `f= zero_point + static_cast<int32_t>(Round(value / scale));`

I've added it as a TODO to update the pt mobile implementation to be on par with fbgemm.